CMakeLists.txt

```diff
@@ -7,7 +7,14 @@ include(bin/dsn.cmake)
 #project(dsn VERSION 1.0 LANGUAGES C CXX)
 project(dsn C CXX)
 set(DSN_BUILD_RUNTIME TRUE)
-set(GIT_SOURCE_FILE "ext/gitsource.cmake")
+
+if(DSN_GIT_SOURCE STREQUAL "xiaomi")
```

Comment:
It is best to change it to something like follows so that others may also use this facility.

if (!defined(DSN_GIT_SOURCE) OR DSN_GIT_SOURCE STREQL "")
   set src = github
endif

set(GIT_SOURCE_FILE "ext/gitsource."$(src))